### ISSUE TITLE
pip_requirements `--extra-index-url` not used as registry URL

### ISSUE THREAD

**Issue by simonhaenisch** [renovatebot/renovate#18028]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.164.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

I think this used to work but not totally sure. I found PR #15868 which changed the behavior of URLs specified in `--extra-index-url` in `requirements.txt` to be put into `additionalRegistryUrls` instead of `registryUrls`. That apparently wasn't working because then there was PR #16334 to fix additional registry URLs not being used. However it still doesn't seem to work.

Renovate doesn't seem to be using the additional URLs when looking for updates. In the debug logs I couldn't find any requests to our extra registry (looking at `urls` from the "http statistics" log), and Renovate couldn't find the update I was expecting. Instead it just added a warning "failed to look up dependency" (which was caused by the package not existing in the pypi.org registry).

Adding the extra URL to `python.registryUrls` in our config is a working work-around.

```js
{
  "python": {
    "registryUrls": [
      "https://pypi.org/pypi/",
      "https://artifactory.company.net/artifactory/api/pypi/our-repo/simple/"
    ]
  },
  // ...
}
```

After this I could see requests being made to our registry in the http statistics debug log, and Renovate was able to find the update.

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment]
Can you provide an example `requirements.txt` in a repo which reproduces this? You can switch out your private registry for a dummy name before sharing.

**Comment by simonhaenisch** [comment]
```
--extra-index-url=https://artifactory.company.net/artifactory/api/pypi/l-pypi/simple/

custom-package-from-our-registry==0.20.0
PyYAML==5.4.1
jsonschema==4.16.0
```

Something like that, sorry I don't have a public repo to share... I will try to create a reproduction however I'm not that familiar with setting up Python projects.

**Comment by rarkins** [comment]
And what would be the expected behavior from this? e.g. try PyPI first and if not found then try the extra index URL after that?

**Comment by simonhaenisch** [comment]
Yes, exactly. (I think if I'd want it to try my registry first, I assume I'd set it up as `--index-url` instead)

**Comment by simonhaenisch** [comment]
Closing this for now because we updated renovate to the latest version today and I'm hoping that this fixes it 🤞 (even though our previous version supposedly had the fixes I mentioned)
